### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yty25**

### CODE REVIEW

File: linux_sql/README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
-# Linux Cluster Monitoring Agent
-This project is under development. Since this project follows the GitFlow, the final work will be merged to the master branch after Team Code Team.
+## Introduction
+The main task about this project is we want to record the hardware specification of 10 different nodes and monitor nodes resource usages such as memory
+used in realtime. We collect data using shell scripts and store the data in postgreSQL database. This help the LCA team to monitor the hardware status
+of each node and be able to analyze those data in order to prevent potential issues.  
+  
+## Architecture and design
+![my image](./assets/arch.png)  
+1)database
```
Comment (Jarvis-deprecated-account) [inline linux_sql/README.md]: use md

File: linux_sql/scripts/host_usage.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (Jarvis-deprecated-account) [inline linux_sql/scripts/host_usage.sh:1]: remove

File: linux_sql/scripts/host_usage.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+timestamp=$(echo "$vmstat_out" | tail -n 1 | awk '{printf("%s %s",  $(NF-1), $NF)}')
+host_name=$(hostname -f)
+memory_free=$(echo "$vmstat_out" | egrep "free memory" | awk '{print$1}')
+cpu_idel=$(echo "$vmstat_out" | tail -n 1 | awk '{print $(NF-2)}')
+cpu_kernel=$(echo "$vmstat_out" |awk 'END {print $14}')
+disk_io=$(echo "$vmstat_out" | egrep "inprogress" | awk '{print $1 * 1000}')
+disk_available=$(echo "$(df -BM /)" | tail -n 1 | awk '{print substr($4, 0, length($4)-1)}')
+
+#insert the data into the psql database.
```
Comment (Jarvis-deprecated-account) [inline linux_sql/scripts/host_usage.sh:35]: fix string format

File: linux_sql/scripts/psql_docker.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	if [[ "$(docker volume ls | egrep "pgdata1$" | wc -l)" == 0 ]];
+	then
+		# delete the container named jrvs-psql if there is one
+		if [[ "$(docker ps -a -f name=jrvs-psql | wc -l)" == 2 ]];
+		then
+			docker stop jrvs-psql
+		fi
+		# create the volume and container
+		docker volume create pgdata
+		docker run --name jrvs-psql -e POSTGRES_PASSWORD=$PGPASSWORD -d -v pgdata:/var/lib/postgresql/data -p 5432:5432 postgres
```
Comment (Jarvis-deprecated-account) [inline linux_sql/scripts/psql_docker.sh:39]: call this once

File: linux_sql/scripts/psql_docker.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	systemctl status docker && docker stop jrvs-psql
+	exit 0
+else
+	echo "invalid arguments"
+	echo "usage: psql_docker.sh start|stop [password for database]"
+	exit 1
+fi
+
+exit 0
+
```
Comment (Jarvis-deprecated-account) [inline linux_sql/scripts/psql_docker.sh]: remove empty lines

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by yty25 — COMMENTED** [review]
Resolved

**Pull request merged**